java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightServer.java

```diff
@@ -317,6 +340,15 @@ public Builder useTls(final File certChain, final File key) throws IOException {
       return this;
     }
 
+    /**
+     * Enable Client Verification via mTLS on the server.
+     * @param mTlsCACert The CA certificate to use for verifying clients.
+     */
+    public Builder useMTlsClientVerification(final File mTlsCACert) throws IOException {
+      this.mTlsCACert = new FileInputStream(mTlsCACert);
```

Comment:
If the user calls this method repeatedly we will leak File handles. Need to close the current one if it exists.

Reply:
Hey @jduo - I'm not a very good Java developer - but I'm learning (thanks for your patience).  Would it be like this?
```java
    public Builder useMTlsClientVerification(final File mTlsCACert) throws IOException {
      if (this.mTlsCACert != null) {
            this.mTlsCACert.close();
        }
      this.mTlsCACert = new FileInputStream(mTlsCACert); 
```
Thanks for your help!

Reply:
Yes this would work. However I'm leaning towards deferring opening mTlsCACert until the connection is made. It's odd for builders to have more logic than setting fields.

Reply:
@jduo just for my knowledge and to clarify a point, would it be more easier if we use try-with-resources here?

Reply:
hi @jduo - this is the FlightServer - I'd rather not change the overall design of the current class's builder.  It already uses this approach for `useTls` - and I think my change is consistent with that.

Reply:
@vibhatha , try-with-resources wouldn't help here since we need the stream to have a longer lifetime than this builder function. This is more of a case of tracking ownership.

Reply:
Got it, thanks @jduo

Reply:
@jduo - I've added small changes that detect if the InputStream attributes are not null, and if so - closes them.  Could you please re-review?

Reply:
Thanks @prmoore77 minor comment about another possible error scenario in useTls() that applies here.